docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -43,13 +43,40 @@ OR
 }
 ```
 
-`"null"` is not considered for type casting logic, only for the nullability constraint. Type lookup will continue down the table.
+`"null"` is not considered for type casting logic, only for the nullability constraint. Type lookup will continue down
+the table. `"null"`'s position in a list does not matter. Nullability constraints only apply to the first major of the 
```

Comment:
I am not sure I understand what it means that nullability constraints only apply to the first major. Does it mean that adding or removing `null` in minor versions has no effect?

Reply:
>  Does it mean that adding or removing null in minor versions has no effect?

Correct. It is impossible to add NOT NULL column to existing table. 
Loosening NULL condition theoretically possible but we do no do that.

Reply:
it also has affect on validation in enrich, but not for warehouse schemas.